extendeddatacrossword.go

```diff
@@ -145,7 +145,11 @@ func (eds *ExtendedDataSquare) solveCrosswordRow(
 	// Check that newly completed orthogonal vectors match their new merkle roots
 	for c := 0; c < int(eds.width); c++ {
 		col := eds.col(uint(c))
-		if noMissingData(col) {
+		if col[r] != nil {
+			continue // not newly completed
+		}
+		col[r] = rebuiltShares[c]
```

Comment:
[question]
I think we're inserting some of the rebuilt shares twice, once here and then a second time on line 161. Should we instead insert the rebuilt shares before performing this step?

same goes for the change below

Reply:
I don't think so, because the EDS is only meant to contain the "last known" good square, so we can't insert the cell into the EDS until we know it's good.

Reply:
that makes sense!